samples/triangle/main.c

```diff
@@ -191,59 +190,39 @@ static void init_shader(void)
     p = pb_begin();
 
     /* Set run address of shader */
-    pb_push1(p, NV097_SET_TRANSFORM_PROGRAM_START, 0);
-    p += 2;
+    p = xgu_set_transform_program_start(p, 0);
 
     /* Set execution mode */
-    pb_push1(p, NV097_SET_TRANSFORM_EXECUTION_MODE,
-        MASK(NV097_SET_TRANSFORM_EXECUTION_MODE_MODE, NV097_SET_TRANSFORM_EXECUTION_MODE_MODE_PROGRAM)
-        | MASK(NV097_SET_TRANSFORM_EXECUTION_MODE_RANGE_MODE, NV097_SET_TRANSFORM_EXECUTION_MODE_RANGE_MODE_PRIV));
-    p += 2;
-
-    pb_push1(p, NV097_SET_TRANSFORM_PROGRAM_CXT_WRITE_EN, 0);
-    p += 2;
+    p = xgu_set_transform_execution_mode(p, XGU_PROGRAM, XGU_RANGE_MODE_PRIVATE);
+    p = xgu_set_transform_program_cxt_write_enabled(p, false);
 
     /* Set cursor and begin copying program */
-    pb_push1(p, NV097_SET_TRANSFORM_PROGRAM_LOAD, 0);
-    p += 2;
-
-    for (i=0; i<sizeof(vs_program)/8; i++) {
-        pb_push(p++, NV097_SET_TRANSFORM_PROGRAM, 4);
-        memcpy(p, &vs_program[i*4], 4*4);
-        p+=4;
-    }
+    p = xgu_set_transform_program_load(p, 0);
+    p = xgu_set_transform_program(p, vs_program, sizeof(vs_program)/8);
```

Comment:
Probably should be `/16` - the original code is very weird, and I believe this is a bug